autoPyTorch/api/base_task.py

```diff
@@ -551,89 +551,150 @@ def _do_dummy_prediction(self, num_run: int) -> None:
+        """
 
         # Mypy Checkings -- Traditional prediction is only called for search
         # where the following objects are created
         assert self._metric is not None
         assert self._logger is not None
         assert self._dask_client is not None
 
-        self._logger.info("Starting to create dummy predictions.")
-
         memory_limit = self._memory_limit
         if memory_limit is not None:
             memory_limit = int(math.ceil(memory_limit))
         available_classifiers = get_available_classifiers()
-        dask_futures = list()
-        time_for_traditional_classifier_sec = int(time_for_traditional / len(available_classifiers))
+        dask_futures = []
+
+        total_number_classifiers = len(available_classifiers) + num_run
         for n_r, classifier in enumerate(available_classifiers, start=num_run):
+
+            # Only launch a task if there is time
             start_time = time.time()
-            scenario_mock = unittest.mock.Mock()
-            scenario_mock.wallclock_limit = time_for_traditional_classifier_sec
-            # This stats object is a hack - maybe the SMAC stats object should
-            # already be generated here!
-            stats = Stats(scenario_mock)
-            stats.start_timing()
-            ta = ExecuteTaFuncWithQueue(
-                backend=self._backend,
-                seed=self.seed,
-                metric=self._metric,
-                logger_port=self._logger_port,
-                cost_for_crash=get_cost_of_crash(self._metric),
-                abort_on_first_run_crash=False,
-                initial_num_run=num_run,
-                stats=stats,
-                memory_limit=memory_limit,
-                disable_file_output=True if len(self._disable_file_output) > 0 else False,
-                all_supported_metrics=self._all_supported_metrics
-            )
-            dask_futures.append((classifier, self._dask_client.submit(ta.run, config=classifier,
-                                                                      cutoff=time_for_traditional_classifier_sec)))
+            if time_left >= func_eval_time_limit_secs:
+                self._logger.info(f"{n_r}: Started fitting {classifier} with cutoff={func_eval_time_limit_secs}")
+                scenario_mock = unittest.mock.Mock()
+                scenario_mock.wallclock_limit = time_left
+                # This stats object is a hack - maybe the SMAC stats object should
+                # already be generated here!
+                stats = Stats(scenario_mock)
+                stats.start_timing()
+                ta = ExecuteTaFuncWithQueue(
+                    backend=self._backend,
+                    seed=self.seed,
+                    metric=self._metric,
+                    logger_port=self._logger_port,
+                    cost_for_crash=get_cost_of_crash(self._metric),
+                    abort_on_first_run_crash=False,
+                    initial_num_run=n_r,
```

Comment:
It looks the definitions of `num_run` and `n_r` changed. `num_run` and `n_r` are identifiers of algorithms, right?

Reply:
yes exactly, and they changed because there was a bug in that two machine learning algorithms previously had the same number. So two algorithm where written to the same files in disk,